api/client_test.go

```diff
@@ -47,46 +75,347 @@ func Test_DefaultClientConfig_WithEnvVars(t *testing.T) {
+			q := &QueryParam{
+				IncludeEvents: tt.fields.IncludeEvents,
+				Status:        tt.fields.Status,
+				Run:           tt.fields.Run,
+			}
+			assert.Equalf(t, tt.want, q.Encode(), "Encode()")
+		})
+	}
+}
+
+func Test_StatusClient_Overall_BadResponse(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, "foo")
+		assert.NoError(t, err)
+	}))
+
+	defer server.Close()
+
+	clientConfig := BaseClientConfig()
+	clientConfig.URL = server.URL
+	c, err := NewClient(clientConfig, nil)
+	assert.NoError(t, err)
+
+	o, err := c.Status().Overall()
+	assert.Error(t, err)
+	assert.EqualValues(t, OverallStatus{}, o)
+}
+
+func Test_StatusClient_Overall_Error_BadResponse(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+	}))
+
+	defer server.Close()
+
+	clientConfig := BaseClientConfig()
+	clientConfig.URL = server.URL
+	c, err := NewClient(clientConfig, nil)
+	assert.NoError(t, err)
+
+	o, err := c.Status().Overall()
+	assert.Error(t, err)
+	assert.EqualValues(t, OverallStatus{}, o)
+}
+
+func Test_StatusClient_Overall_Error_WithErrorMessage(t *testing.T) {
+	e := &ErrorResponse{Error: &ErrorObject{Message: "foo"}}
+	bytes, err := json.Marshal(e)
+	assert.NotEmpty(t, bytes)
+	assert.NoError(t, err)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+		_, err := w.Write(bytes)
+		assert.NoError(t, err)
+	}))
+
+	defer server.Close()
+
+	clientConfig := BaseClientConfig()
+	clientConfig.URL = server.URL
+	c, err := NewClient(clientConfig, nil)
+	assert.NoError(t, err)
+
+	o, err := c.Status().Overall()
+	assert.Error(t, err)
+	assert.EqualValues(t, OverallStatus{}, o)
+}
+
+func Test_StatusClient_Overall_Error_NoErrorMessage(t *testing.T) {
+	e := &ErrorResponse{}
+	bytes, err := json.Marshal(e)
+	assert.NotEmpty(t, bytes)
+	assert.NoError(t, err)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+		_, err := w.Write(bytes)
+		assert.NoError(t, err)
+	}))
+
+	defer server.Close()
+
+	clientConfig := BaseClientConfig()
+	clientConfig.URL = server.URL
+	c, err := NewClient(clientConfig, nil)
+	assert.NoError(t, err)
+
+	o, err := c.Status().Overall()
+	assert.Error(t, err)
+	assert.EqualValues(t, OverallStatus{}, o)
+}
+
+func Test_StatusClient_Overall(t *testing.T) {
+	expectedOverallStatus := OverallStatus{TaskSummary: TaskSummary{
+		Status:  StatusSummary{Successful: 1},
+		Enabled: EnabledSummary{True: 2},
+	}}
+
+	bytes, err := json.Marshal(&expectedOverallStatus)
+	assert.NotEmpty(t, bytes)
+	assert.NoError(t, err)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err = fmt.Fprintf(w, string(bytes))
```

Comment:
Similar linting error here with `fmt.Fprintf`

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/commit/808db068de5e422e1b0ce82a0f8a0c8374509a20